client/client.lua

```diff
@@ -241,10 +241,18 @@ exports['qtarget']:AddTargetModel(impoundPeds, {
 AddStateBagChangeHandler('vehicleData', nil, function(bagName, key, value, _unused, replicated)
     if not value then return end
     local entNet = bagName:gsub('entity:', '')
-    while not NetworkDoesEntityExistWithNetworkId(tonumber(entNet)) do Wait(0) end
+    local timer = GetGameTimer()
+    while not NetworkDoesEntityExistWithNetworkId(tonumber(entNet)) and GetGameTimer() - timer < 10000 do Wait(0) end
     local vehicle = NetToVeh(tonumber(entNet))
-    lib.setVehicleProperties(vehicle, json.decode(value.vehicle))
-    TriggerServerEvent('luke_garages:ChangeStored', value.plate, false, nil)
+    local timer = GetGameTimer()
+    while NetworkGetEntityOwner(vehicle) ~= PlayerId() and GetGameTimer() - timer < 10000 do
+        Wait(0)
+    end
```

Comment:
If the player isn't the entity owner, wouldn't this cause an infinite loop?

Reply:
oops, will fix now.